demonstrations/tutorial_stochastic_parameter_shift.py

```diff
@@ -0,0 +1,397 @@
+
+i) The circuit's gates have free parameters
+ii) Expectation values of measurements are built up from samples
+
+These two ingredients allow one circuit to actually represent an entire *family of circuits*. 
+An objective function---encapsulating some problem-specific goal---is built from the expectation values, 
+and the circuit's free parameters are progressively tuned to optimize this function. 
+At each step, the circuit has the same gate layout, but slightly different parameters, making 
+this approach promising to run on constrained near-term devices.
+
+But how should we actually update the circuit's parameters to move us closer to a good output? 
+Borrowing a page from classical optimization and deep learning, we can use 
+`gradient descent <https://en.wikipedia.org/wiki/Gradient_descent>`_. 
+In this general-purpose method, we compute the derivative of a (smooth) function :math:`f` with 
+respect to its parameters :math:`\theta`, i.e., its gradient :math:`\nabla_\theta f`. 
+Since the gradient always points in the direction of steepest ascent/descent, if we make small updates 
+to the parameters according to
+
+.. math::
+
+    \theta \rightarrow \theta - \eta \nabla_\theta f,
+    
+we can iteratively progress to lower and lower values of the function.
+
+The Parameter-Shift Rule
+------------------------
+
+In the quantum case, the expectation value of a circuit with respect to an measurement operator 
+:math:`\hat{C}` depends smoothly on the the circuit's gate parameters :math:`\theta`. We can write this
+expectation value as :math:`\langle \hat{C}(\theta)\rangle`. This means that the derivatives 
+:math:`\nabla_\theta \langle \hat{C} \rangle` exist and gradient descent can be used. 
+
+Before digging deeper, we will first set establish some basic notation. For simplicity, though a circuit 
+may contain many gates, we can concentrate on just a single gate :math:`\hat{U}` that we want to differentiate
+(other gates will follow the same pattern).
+
+.. figure:: ../demonstrations/stochastic_parameter_shift/quantum_circuit.png
+    :align: center
+    :width: 90%
+
+All gates appearing before :math:`\hat{U}` can be absorbed into an initial state preparation 
+:math:`\vert \psi_0 \rangle`, and all gates appearing after :math:`\hat{U}` can be absorbed with the measurement
+operator :math:`\hat{C}` to make a new effective measurement operator :math:`\hat{A}`.
+The expectation value :math:`\hat{A}` in the simpler one-gate circuit is identical to 
+the expectation value :math:`\hat{C}` in the larger circuit.
+
+We can also write any unitary gate in the form
+
+.. math::
+
+    \hat{U}(\theta) = e^{i\theta\hat{V}},
+    
+where :math:`\hat{V}` is the Hermitian *generator* of the gate :math:`\hat{U}`.
+
+Now, how do we actually obtain the numerical values for the gradient necessary for gradient descent? 
+
+This is where the parameter-shift rule [#li2016]_, [#mitarai2018]_, [#schuld2018]_ enters the story. 
```

Comment:
Nitpick: looks slightly better without the commas, as this matches revtex and wikipedia
```suggestion
This is where the parameter-shift rule [#li2016]_ [#mitarai2018]_ [#schuld2018]_ enters the story. 
```